### ISSUE TITLE
error serializing property with `oneOf` constraint

### ISSUE THREAD

**Issue by restjohn** [byu-oit/openapi-enforcer#58]
I have an OpenAPI document that includes a schema like the following snippet.
```yaml
components:
  schemas:
    TypeA:
    properties:
      id:
        type: string
      related:
        oneOf:
          - $ref: '#/components/schemas/TypeB/properties/id'
          - $ref: '#/components/schemas/TypeB
```
When using OpenAPI Enforcer Middleware and returning a document from a request like the following
```json
{ "id": "abc123", "related": "xyz987" }
```
OpenAPI Enforcer throws the following exception:
```
TypeError: Cannot assign to read only property '0' of object '[object String]'
    at Function.assign (<anonymous>)
    at runSerialize (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:97:24)
    at Object.keys.forEach.key (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:120:34)
    at Array.forEach (<anonymous>)
    at runSerialize (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:118:32)
    at Object.keys.forEach.key (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:122:34)
    at Array.forEach (<anonymous>)
    at runSerialize (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:118:32)
    at Object.keys.forEach.key (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:120:34)
    at Array.forEach (<anonymous>)
    at runSerialize (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/schema/serialize.js:118:32)
    at Schema.serialize (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/enforcers/Schema.js:193:24)
    at Operation.response (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer/src/enforcers/Operation.js:381:64)
    at ServerResponse.res.send (/Users/stjohnr/sandbox/mage/mage-server/node_modules/openapi-enforcer-middleware/index.js:194:55)
    at /Users/stjohnr/sandbox/mage/mage-server/plugins/mage-manifold/index.ts:50:18
    at Generator.next (<anonymous>)
    at fulfilled (/Users/stjohnr/sandbox/mage/mage-server/plugins/mage-manifold/index.ts:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
rather than the reference to the other schema property.

The trouble is this line https://github.com/byu-oit/openapi-enforcer/blob/5cafccf34cb0104068b35e21100921f424f083cf/src/schema/serialize.js#L97, which is attempting to use `Object.assign()` to assign a property to the string value of the `"xyz987"`.

**Comment by Gi60s** [comment]
Thank you for the issue. I'll get working on a fix. This should be done in the next few hours.

**Comment by Gi60s** [comment]
Fixed and published to NPM as version `1.8.2`. Thank you for the issue and for identifying where it was occurring.

**Comment by restjohn** [comment]
No problem.  Thanks for the quick response!
